### PR TITLE
EOS-19399: Create design RFC for AUX multipools layout generation

### PR DESCRIPTION
**Pull request by suvratjoshi**
Need to create design document RFC, for generation of layout parameters for auxiliary (AUX) pools, during multipools cluster bootstrap.

-----
[View rendered rfc/22/README.md](https://github.com/suvratjoshi/cortx-hare/blob/e19399/rfc/22/README.md)
[View rendered rfc/README.md](https://github.com/suvratjoshi/cortx-hare/blob/e19399/rfc/README.md)

### CODE REVIEW

File: rfc/22/README.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    disk_refs:
+      - { path: /dev/mpath1, node: srvnode-1 }
+      - { path: /dev/mpath2, node: srvnode-2 }
+      - { path: /dev/mpath3, node: srvnode-3 }
+      - { path: /dev/mpath4, node: srvnode-4 }
+    data_units: 2
+    parity_units: 2
+    allowed_failures: { site: 0, rack: 0, encl: 0, ctrl: 0, disk: 2 }
+```
+We have to calculate all possible combinations (without repetition) of the four surviving srvnode's out of the total six nodes, and cfgen needs to generate pool configuration with reduced 2+2 layout for each of them internally. There will 15 of them, according to the formula: C(p,k) = p! / ( k! * (p - k)! ). In our case p := 6 and k := 4:
```
Comment (mssawant) [inline rfc/22/README.md:52]: /There will 15 of them/ There will be 15 of them

**Comment by mssawant** [comment]
@suvratjoshi, please fix the DCO.

File: rfc/22/README.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+## Abstract
+
+The multipools feature gives cluster an ability to have multiple pools (sets) of the disk/storage resources and IO's can be done on the desired pool.
+The auxiliary pools feature is required for creating auxialiry pool sets, one of which could be used upon failure of one or more disk resources.
+
+## Design
+
+If some disk resources in a cluster fail for some reason temporarily (because of reasons like reboot or transient network error etc.), then it is expected that cluster still works properly with reduced pool layout for failure handling. Let's say the initial pool layout configuration is 4+2, and a storage set has 6 data volumes. If 1-2 of those data volumes fail, the pool is switched to 2+2 layout on the remaining data volumes. There can be several combinations of the 2+2 data volume layouts, depending on which particular two volumes fail. All such combinations need to be pre-populated (auto-generated by Hare from the CDF configuration) by the hctl bootstrap script.
```
Comment (mssawant) [inline rfc/22/README.md:20]: [* suggestion * ] Following example presents a typical Base pool, auxiliary pool and their typical data distribution attributes,

**Comment by knekrasov** [comment]
Yesterday @NeeravChoudhari said that no more RFCs should be created in our source folder. New documents must be created at Confluence ([here is the link](https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/165053761/Hare) to our "folder")

**Comment by suvratjoshi** [comment]
Closing this PR and opening a new PR : https://github.com/Seagate/cortx-hare/pull/1588
Thanks!

**Pull request closed without merging**